controlplane/pkg/nsmd/nsmd_crossconnect_client.go

```diff
@@ -134,6 +162,44 @@ func (client *NsmMonitorCrossConnectClient) ClientConnectionDeleted(clientConnec
 	}
 }
 
+func (client *NsmMonitorCrossConnectClient) endpointConnectionMonitor(endpoint *model.Endpoint, ctx context.Context) {
+	logrus.Infof(endpointLogFormat, endpoint.EndpointName(), "Added")
+	defer func() { logrus.Infof(endpointLogFormat, endpoint.EndpointName(), "Removed") }()
+
+	connected := client.endpointStateManager.IsEndpointAlive(endpoint)
+	err := error(nil)
+
+	for i := 0; i < endpointConnectAttempts && !connected; i++ {
+		select {
+		case <-ctx.Done():
+			logrus.Infof(endpointLogFormat, endpoint.EndpointName(), "Context timeout exceeded...")
+			return
+		case <-time.Tick(endpointLivenessTimeout):
+			connected = client.endpointStateManager.IsEndpointAlive(endpoint)
+		}
+	}
+
+	if connected {
+		logrus.Infof(endpointLogFormat, endpoint.EndpointName(), "Connected")
+	}
+
+	for connected {
```

Comment:
So here, you need to check if it alive forever until it had active connection or alive.

Reply:
It is actually the thing that I am doing here. Probably `IsEndpointAlive()` was not so meaningful name, so I have changed it to `IsEndpointAliveOrHasActiveConnections()` : 43b0835